### PR TITLE
Big HTTP renaming

### PR DESCRIPTION
**Pull request by vinniefalco**
Cosmetic changes only. Documentation is improved and updated.

Doc preview:
http://vinniefalco.github.io/stage/beast/b20/index.html

In particular:
http://vinniefalco.github.io/stage/beast/b20/beast/http/message.html

### CODE REVIEW

**Comment by vinniefalco** [comment]
Reviewers: @HowardHinnant @scottschurr

File: include/beast/http/message.hpp [hunk 1]
```diff
@@ -368,64 +375,60 @@ struct message :
 //------------------------------------------------------------------------------
 
 #if GENERATING_DOCS
-/** Swap two HTTP message headers.
+/** Swap two header objectsd.
```
Comment (scottschurr) [inline include/beast/http/message.hpp]: `objectsd` -> `objects`?

**Comment by vinniefalco** [comment]
quickref.xml is maintained by hand. None of the files in doc/ are automatically generated

File: doc/design.qbk [hunk 2]
```diff
@@ -447,17 +447,16 @@ start. Other design goals:
...
     appended to the sequence when existing allocations are filled. Beast
     does not impose any particular memory management model on callers. The
-    `basic_streambuf` provided by beast supports standard allocators through
-    a template argument. Use the [*DynamicBuffer] that comes with beast,
+    __basic_streambuf__ provided by beast supports standard allocators through
+    a template argument. Use the __streambuf__ that comes with beast,
     customize the allocator if you desire, or provide your own type that
-    meets the
-    [@https://github.com/vinniefalco/Beast/blob/6c8b4b2f8dde72b01507e4ac7fde4ffea57ebc99/include/beast/basic_streambuf.hpp#L21 concept requirements].
+    meets the requirements of __DynamicBuffer__.
```
Comment (vinniefalco) [inline doc/design.qbk]: No, DynamicBuffer is the concept. basic_streambuf is a type.

File: doc/design.qbk [hunk 3]
```diff
@@ -447,17 +447,16 @@ start. Other design goals:
...
+    Beast comes with the class __basic_streambuf__, an efficient
+    implementation of the __DynamicBuffer__ concept which makes use of multiple
     allocated octet arrays. If an incoming message is broken up into
     multiple pieces, no reallocation occurs. Instead, new allocations are
     appended to the sequence when existing allocations are filled. Beast
     does not impose any particular memory management model on callers. The
-    `basic_streambuf` provided by beast supports standard allocators through
-    a template argument. Use the [*DynamicBuffer] that comes with beast,
+    __basic_streambuf__ provided by beast supports standard allocators through
+    a template argument. Use the __streambuf__ that comes with beast,
```
Comment (vinniefalco) [inline doc/design.qbk]: will change

File: include/beast/http/basic_fields.hpp [hunk 4]
```diff
@@ -21,35 +21,35 @@
 namespace beast {
 namespace http {
 
-/** A container for storing HTTP headers.
+/** A container for storing HTTP header fields.
 
     This container is designed to store the field value pairs that make
-    up the headers and trailers in a HTTP message. Objects of this type
+    up the fields and trailers in a HTTP message. Objects of this type
     are iterable, which each element holding the field name and field
```
Comment (vinniefalco) [inline include/beast/http/basic_fields.hpp]: will fix

File: include/beast/http/message.hpp (hunk 1 above)
Comment (vinniefalco) [inline include/beast/http/message.hpp]: will fix

**Comment by HowardHinnant** [comment]
👍  Looks good to me.  Passes tests on macOS.  We get warnings coming out of asio, but there's not much we can do about that.

**Pull request closed without merging**